tests/models/test_metatensor.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import torch
+
+from tests.models.conftest import (
+    consistency_test_simstate_fixtures,
+    make_model_calculator_consistency_test,
+    make_validate_model_outputs_test,
+)
+
+
+try:
+    from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+    from metatrain.utils.io import load_model
+
+    from torch_sim.models.metatensor import MetatensorModel
+except ImportError:
+    pytest.skip("Metatensor not installed", allow_module_level=True)
+
+
+@pytest.fixture
+def dtype() -> torch.dtype:
+    """Fixture to provide the default dtype for testing."""
+    return torch.float32
+
+
+@pytest.fixture
+def metatensor_calculator(device: torch.device):
+    """Load a pretrained metatensor model for testing."""
+    return MetatensorCalculator(
+        model=load_model(
+            "https://huggingface.co/lab-cosmo/pet-mad/resolve/main/models/pet-mad-latest.ckpt"
+        ).export(),
+        device=device,
+    )
+
+
+@pytest.fixture
+def metatensor_model(device: torch.device) -> MetatensorModel:
+    """Create an MetatensorModel wrapper for the pretrained model."""
+    return MetatensorModel(
+        model="pet-mad",
+        device=device,
+    )
+
+
+def test_metatensor_initialization(device: torch.device) -> None:
+    """Test that the metatensor model initializes correctly."""
+    model = MetatensorModel(
+        model="pet-mad",
+        device=device,
+    )
+    assert model._device == device  # noqa: SLF001
+    assert model._dtype == torch.float32  # noqa: SLF001
```

Comment:
I'd test the public attributes `model.device` and `model.dtype` instead of the private versions.